test/blackbox/common/BlackboxTestsLivelinessQos.cpp

```diff
@@ -1946,6 +1948,54 @@ TEST(LivelinessTests, Detect_Deadlock_ManualByParticipant_Intraprocess)
     // Test failure is due to timeout
 }
 
+// Regression test of Refs #20584, github issue #4373
+TEST(LivelinessTests, Reader_Successfully_Asserts_Liveliness_on_a_Disconnected_Writer)
+{
+    // Create a TestTransport to simulate a network shutdown (Ctrl+C)
+    auto testTransport = std::make_shared<eprosima::fastdds::rtps::test_UDPv4TransportDescriptor>();
+
+    // Create two writer participants
+    PubSubWriter<HelloWorldPubSubType> writer_1(TEST_TOPIC_NAME);
+    PubSubWriter<HelloWorldPubSubType> writer_2(TEST_TOPIC_NAME + "2");
```

Comment:
Shouldn't we check with the same topic name?

Reply:
Not necessarily (although I think it would also work. It is enough having a pair of reader/writer matched in different participants). However I tried to stick to the scenario depicted in the issue and I think that distinguishing topic names fits better with the `PubSubPaarticipant` API so we do not try creating two topics with the same same when we `init_subscriber()`